src/Datasource/ConnectionInterface.php

```diff
@@ -79,4 +72,61 @@ public function disableConstraints(callable $operation);
      * @return bool
      */
     public function logQueries($enable = null);
+
+    /**
+     * Sets a logger
+     *
+     * @param \Cake\Database\Log\QueryLogger $logger Logger object
+     * @return $this
+     */
+    public function setLogger($logger);
```

Comment:
Would be nice to have at least a basic interface here to contract against instead of using a concrete and specific class.